libminifi/include/core/StreamAppendLock.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+
+namespace org::apache::nifi::minifi::core {
+
+class StreamAppendLock {
```

Comment:
Since its derived class `ContentStreamAppendLock` arrives in the file of the associated `ContentRepository`, wouldn't this also be better next to `StreamManager`?

Reply:
do you mean an inner class?

Reply:
Either that or just a normal top level class in the same file.

Reply:
done